tests/client/core/api/test_api.js

```diff
@@ -88,9 +93,9 @@ describe('api', () => {
           {headers: {}, method: 'get'})
         .once()
         .returns(Promise.resolve({ok: false}));
-      return api.fetchAddon({slug: 'foo'}).then(
-        () => assert.fail(null, null, 'expected API call to fail'),
-        (error) => assert.equal(error.message, 'Error calling API'));
+      return api.fetchAddon({slug: 'foo'})
+        .then(helpers.unexpectedSuccess,
```

Comment:
I'm not a huge fan of this style of indenting. If we're going to break I prefer it happening earlier. This is mostly because I find it finicky to line up the code with the `(` instead of just indenting it. It isn't too bad with a `then()` but with longer function names it becomes a real pain. I could likely be persuaded the other way but I find more and more I prefer the early break now.

```js
      return api.fetchAddon({slug: 'foo'})
        .then(
          helpers.unexpectedSuccess,
          (error) => assert.equal(error.message, 'Error calling API'));